xbstrap/schema.yml

```diff
@@ -236,6 +236,8 @@ properties:
                     type: string
                 'disable_shallow_fetch':
                     type: boolean
+                'recursive_sources':
+                    type: boolean
```

Comment:
Instead of adding another `recursive_sources` key, we should handle this similarly to the `tools_required:` case, namely by allowing entries to be dicts instead of strings:
```yml
sources_required:
  - name: foobar
    recursive: true
```